cql3/statements/select_statement.hh

```diff
@@ -341,7 +338,7 @@ public:
 private:
     future<exceptions::coordinator_result<service::storage_proxy_coordinator_query_result>>
     do_query(
-            const locator::node* this_node,
+            std::optional<locator::host_id> this_node,
```

Comment:
also, `locator::host_id` can use optimized_optional, or be passed as naked since it has a bool operator and a distinct null value, similar to a `node*`

Reply:
Ok, will send v2, with the optional removed.

Reply:
Done in v2.